### PR TITLE
Toast text color

### PR DESCRIPTION
**Pull request by signedav**
While I quickly checked it out according input here https://github.com/opengisch/QField/pull/1953#issuecomment-853029223 I decided to share it to discuss.

toast-text-color

### CODE REVIEW

**Comment by nirvn** [comment]
Why not using error red instead of warning orange for error messages?

**Comment by nirvn** [comment]
I like the idea btw :)

File: src/qml/qgismobileapp.qml [hunk 1]
```diff
@@ -2163,8 +2163,9 @@ ApplicationWindow {
 
       background: Rectangle { color: "transparent" }
 
-      function show(text) {
+      function show(text, type) {
           toastMessage.text = text
+          toastContent.type = type
```
Comment (suricactus) [inline src/qml/qgismobileapp.qml:2168]:
```suggestion
          toastContent.type = type || "info"
```

**Comment by signedav** [comment]
I prefer text color over bg color
toast-bg-color

**Comment by signedav** [comment]
> Why not using error red instead of warning orange for error messages?

We need to define what is an error and what is not. For me "Constraints not valid" is a warning. "Could not save feature" is an error. But this is not always that clear I guess.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 94d092c53907a680e1006f963c20e09ac02bc1ba:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-1bc5bed8f29b0f1d2d52f093c9a4bffc397bbc18-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-1bc5bed8f29b0f1d2d52f093c9a4bffc397bbc18-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-1bc5bed8f29b0f1d2d52f093c9a4bffc397bbc18-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-1bc5bed8f29b0f1d2d52f093c9a4bffc397bbc18-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a042bef974afa27a58649330107613f5295ff596:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-be331abfeab847167b9f4e8d7da6413b4708b622-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-be331abfeab847167b9f4e8d7da6413b4708b622-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-be331abfeab847167b9f4e8d7da6413b4708b622-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-be331abfeab847167b9f4e8d7da6413b4708b622-x86.apk)

**Comment by m-kuhn** [comment]
Could we also prefix it with a colored •, or other icon or outline or ... ?
I think it would increase readability to keep the text itself white.

**Comment by signedav** [comment]
Indicator looks better than I thought:
indicator

But I guess I like this most:
frame

Here with both, but I don't like it:
both

**Comment by suricactus** [comment]
I like the indicator only the most from the whole thread.

**Comment by signedav** [comment]
Yeah, I like the indicator visually but I don't know if it brings the effect it should.

**Comment by m-kuhn** [comment]
I think it will work for both, people who work with it a lot (and are conditioned and more annoyed by attention grabbing) and people who work with it sometimes and need a visual cue and will see a color.

**Comment by signedav** [comment]
Okay, then I'll make those indicators ... :bowling:  

Look. Indicators:
indicators_made_for_the_guys

:wink: 
;-)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 9ef4dca1833834e2fb16521e14791222641413a1:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-fc70426e071ce15f686c9db1bbd1767c4b9ba963-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-fc70426e071ce15f686c9db1bbd1767c4b9ba963-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-fc70426e071ce15f686c9db1bbd1767c4b9ba963-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-fc70426e071ce15f686c9db1bbd1767c4b9ba963-x86.apk)

File: src/qml/geometry_editors/ReshapeToolBar.qml [hunk 2]
```diff
@@ -58,7 +58,7 @@ VisibilityFadingRow {
             var result = GeometryUtils.reshapeFromRubberband(featureModel.currentLayer, featureModel.feature.id, rubberbandModel)
             if ( result !== QgsGeometryStatic.Success )
             {
-                displayToast( qsTr( 'An error when reshaping the geometry' ) );
+                displayToast( qsTr( 'An error when reshaping the geometry' ), 'error' );
```
Comment (m-kuhn) [inline src/qml/geometry_editors/ReshapeToolBar.qml:61]:
```suggestion
                displayToast( qsTr( 'The geometry could not be reshaped' ), 'error' );
```

File: src/qml/geometry_editors/FillRingToolBar.qml [hunk 3]
```diff
@@ -66,15 +66,15 @@ VisibilityFadingRow {
       if ( result !== QgsGeometryStatic.Success )
       {
         if ( result === QgsGeometryStatic.AddRingNotClosed )
-          displayToast( qsTr( 'Error: the input ring is not closed' ) );
+          displayToast( qsTr( 'Error: the input ring is not closed' ), 'error' );
```
Comment (m-kuhn) [inline src/qml/geometry_editors/FillRingToolBar.qml:69]:
```suggestion
          displayToast( qsTr( 'The ring is not closed' ), 'error' );
```

File: src/qml/geometry_editors/FillRingToolBar.qml [hunk 4]
```diff
@@ -66,15 +66,15 @@ VisibilityFadingRow {
...
         else if ( result === QgsGeometryStatic.AddRingNotValid )
-          displayToast( qsTr( 'Error: the input ring is not valid' ) );
+          displayToast( qsTr( 'Error: the input ring is not valid' ), 'error' );
```
Comment (m-kuhn) [inline src/qml/geometry_editors/FillRingToolBar.qml:71]:
```suggestion
          displayToast( qsTr( 'The ring is not valid' ), 'error' );
```

File: src/qml/geometry_editors/FillRingToolBar.qml [hunk 5]
```diff
@@ -66,15 +66,15 @@ VisibilityFadingRow {
...
         else if ( result === QgsGeometryStatic.AddRingCrossesExistingRings )
-          displayToast( qsTr( 'Error: the input ring crosses existing rings (it is not disjoint)' ) );
+          displayToast( qsTr( 'Error: the input ring crosses existing rings (it is not disjoint)' ), 'error' );
```
Comment (m-kuhn) [inline src/qml/geometry_editors/FillRingToolBar.qml:73]:
```suggestion
          displayToast( qsTr( 'The ring crosses existing rings (it is not disjoint)' ), 'error' );
```

File: src/qml/geometry_editors/FillRingToolBar.qml [hunk 6]
```diff
@@ -66,15 +66,15 @@ VisibilityFadingRow {
...
         else if ( result === QgsGeometryStatic.AddRingNotInExistingFeature )
-          displayToast( qsTr( 'Error: the input ring doesn\'t have any existing ring to fit into' ) );
+          displayToast( qsTr( 'Error: the input ring doesn\'t have any existing ring to fit into' ), 'error' );
```
Comment (m-kuhn) [inline src/qml/geometry_editors/FillRingToolBar.qml:75]:
```suggestion
          displayToast( qsTr( 'The ring doesn\'t have any existing ring to fit into' ), 'error' );
```

**Review by m-kuhn — APPROVED** [review]
Thanks, just some unrelated improvements to wording

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3bcedade87ef10c85341fb6c662780982e884a2a:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-b867bd130efe30437bea8551589747691e2dcfd4-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-b867bd130efe30437bea8551589747691e2dcfd4-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-b867bd130efe30437bea8551589747691e2dcfd4-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-b867bd130efe30437bea8551589747691e2dcfd4-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3bcedade87ef10c85341fb6c662780982e884a2a:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-b867bd130efe30437bea8551589747691e2dcfd4-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-b867bd130efe30437bea8551589747691e2dcfd4-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-b867bd130efe30437bea8551589747691e2dcfd4-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-b867bd130efe30437bea8551589747691e2dcfd4-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for aebfa438a31a4efbdf0cc58de601cab8db68be28:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-39247804da6b3a596e96f9f2872321a584523179-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-39247804da6b3a596e96f9f2872321a584523179-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-39247804da6b3a596e96f9f2872321a584523179-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-39247804da6b3a596e96f9f2872321a584523179-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 0e41dca1d90b4656b9cedc33487e26bfb030a6b2:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-e0bd8448c3fabf6e895a5e7226615f450d3e2ef6-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-e0bd8448c3fabf6e895a5e7226615f450d3e2ef6-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-e0bd8448c3fabf6e895a5e7226615f450d3e2ef6-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-e0bd8448c3fabf6e895a5e7226615f450d3e2ef6-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c1384dee44021c08656f6f01057d744e7996bb16:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-c47ad604c268d937685e29cbfe20b6c1ba912a97-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-c47ad604c268d937685e29cbfe20b6c1ba912a97-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-c47ad604c268d937685e29cbfe20b6c1ba912a97-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-c47ad604c268d937685e29cbfe20b6c1ba912a97-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 143aeb4acec3f970d52ad75a4da05c8f35b9b803:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-f078195eea2742fd13592dc622f980a481b78c6c-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-f078195eea2742fd13592dc622f980a481b78c6c-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-f078195eea2742fd13592dc622f980a481b78c6c-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1957-f078195eea2742fd13592dc622f980a481b78c6c-x86.apk)

**Pull request merged**